### ISSUE TITLE
Media Field shows as number

### ISSUE THREAD

**Issue by Jaspervv** [themosis/framework#583]
## Problem
Media Fields show up as a number.
I might be doing something wrong but I couldn't find a Blade template for  the `media` type anywhere, so is this still a **ToDo**?

## Code
I've tried the following:
```php
Field::media('attachments');
```
and
```php
Field::media('attachments', [
    'type' => ['image', 'application']
]);
```
## Result
and both times it shows up in the front-end like a number:
![media-error](https://user-images.githubusercontent.com/35364310/52286984-91724200-2969-11e9-8fb6-4835d0dc4cc3.png)

**Comment by jlambe** [comment]
This is expected. The media field is registering the media ID only. So on the front-end you can fetch any information from your file: url, img tag, ...

**Comment by Jaspervv** [comment]
So from what I understand you must fill in the ID of the image? Wouldn't an image select field be a more user-friendly solution?

For example the way the WordPress costumiser does it:
![image select](https://user-images.githubusercontent.com/35364310/52334148-50c50800-29ff-11e9-9a2a-fcf85b362c73.png)
Which shows a  media select popup:
![image select popup](https://user-images.githubusercontent.com/35364310/52334179-5f132400-29ff-11e9-8015-e3bbd8b98588.png)

**Comment by jlambe** [comment]
The media field is only working when used on metabox or page setting. WordPress does not provide the media library script to the front-end as is, so actually you can't use it on the front-end side from a form for example -> this might be why you see it as a text/number input.

**Comment by Jaspervv** [comment]
Im currently using it inside a custom settings page in the backend, created with the Page Facade: `Themosis\Support\Facades\Page`.
We want to allow clients to upload an image from this page, finding the `ID` of an image and writing it down would be too advanced for them.

Is there currently a way to create an user-friendly way for an image custom field? Or would we have to find another solution?

**Comment by jlambe** [comment]
I responded too quickly. I just test this and the media field is not yet implemented into the settings/page API. As it has been componentized using ReactJS, I haven't yet integrated them for other field contexts. This is an issue actually as Media, Collection, Choice and Color fields should be implemented as well.

**Comment by jlambe** [comment]
@Jaspervv I have decided to not implement the Media, Collection, Choice, Color and other JS dependent fields for the `2.0` release. After some research and because I have componentized the fields as React components, current implementation cannot support those fields.

I have decided to no longer support the default WordPress Settings API and thus provide a custom UI ReactJS based for the settings pages. See issue #605 for details.

As a matter, I may for the 2.0 release throw exception for fields not supported on page settings as a way to alert developers to not use them for now.
